Team54/src/main/java/ISA/Team54/users/controller/PatientController.java

```diff
@@ -0,0 +1,33 @@
+package ISA.Team54.users.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import ISA.Team54.users.model.User;
+import ISA.Team54.users.service.interfaces.PatientService;
+
+@RestController
+@RequestMapping(value = "/api", produces = MediaType.APPLICATION_JSON_VALUE)
+public class PatientController {
+	@Autowired
+	private PatientService patientService;
+	
+	@GetMapping("patientByName/{name}")
+	@PreAuthorize("hasAnyRole('DERMATOLOGIST','PHARMACIST')")
+	public List<User> loadByName(@PathVariable String name){
+		return this.patientService.findByName(name);
+	}
+	
+	@GetMapping("patientBySurname/{surname}")
+	@PreAuthorize("hasAnyRole('DERMATOLOGIST','PHARMACIST')")
+	public List<User> loadBySurname(@PathVariable String surname){
+		return this.patientService.findBySurname(surname);
+	}
```

Comment:
Sta se ovde desava? U patient kontroleru korisnik treba da ima dermatologa i farmaceuta kao rolu?

Reply:
Ne, nego neko ko ima tou rolu moze da pozove ovu metodu

Reply:
tj samo dermatolog i farmaceut mogu da pozovu te metode

Reply:
To mi je jasno, ali zasto zbog cega su u pacijentu role dermatolog i farmaceut. Oni trebaju da dobavljaju pacijente? Zbog cega moraju dobavljati i po imenu i po prezimenu?

Reply:
Odgovorio sam sekund nakon sto si ti odgovorila. Pogledaj drugo pitanje

Reply:
to je pretraga pacijenata po imenu i prezimenu

Reply:
koje drugo pitanje?

Reply:
Mislim da ti ne treba i po jednom i po drugom. Pretpostavljam da ima jedno polje u kojoj se unosi parametar pretrage i ti onda u bazi sa jednim upitom proveris da li ili ime ili prezime ima parametar pretrage u sebi. Hoce biti samo jedno polje?

Reply:
Mislila sam na frontu da stavim dva polja za unos imena i prezimena  u slucaju kad unese jedno da pozove samo jednu metodu kad unese oba verovatno cu dodati jos jednu metodu da se nadje i po imenu i po prezimenu, al to bih naknadno dodala.

Reply:
Zasto ne bi ovako kao sto sam napisao? Cini mi se da je mnogo lakse i intuitivnije.

Reply:
Okej, mozda promenim na frontu logiku i stavim jedno polje, hvala na sugestiji